src/charm.py

```diff
@@ -219,27 +235,74 @@ def self_scraping_job(self):
         """The self-monitoring scrape job."""
         external_url = urlparse(self._external_url)
         metrics_path = f"{external_url.path.rstrip('/')}/metrics"
-        target = (
-            f"{external_url.hostname}{':'+str(external_url.port) if external_url.port else ''}"
-        )
+        internal_url = self._internal_url.replace("http://", "")
+        target = f"{internal_url}"
```

Comment:
- This change (external -> internal) seems to suggest we now expect blackbox exporter to be deployed in the same model as the scraper (grafana agent / prometheus). If that is the case, then we need to clearly document it.
- Should we expect `https://` in addition to `http://`?
- We could clean up be removing the redundant f-string:

```suggestion
        target = re.sub(r"^https?://", "", self._internal_url)
```